### ISSUE TITLE
Feature request : Live view of games 

### ISSUE THREAD

**Issue by schaakverenigingmaastricht** [yo35/rpb-chessboard#196]
Hi I am Helmer Wieringa and work for Tornelo.com and am webmater for schaakverenigingmaastricht,nl 
and have organized several online tournament check for instance https://schaakverenigingmaastricht.nl/winter-series-2020/

We now use for the spring series ,  tornelo.com a tournamentmanagement system for scheduled and arbiter led tournament. 
See https://tornelo.com/chess/orgs/schaakverenigingmaastricht/events/limburg-league-spring-series/lobby

The system has a live broadcast stream in the for of a pgn  file. My question is there any plan to support this scenario. 
I have added an example of such a file: 
**Scenario:** 
-  a player submits a move in the tornelo player 
- the broadcast file is updated 
- rpb notices the update 
- rpb updates a page with a short code 

What would be great if also a multi board would be support  in a responsive way. 

Ok I ask a lot but I have NO and a YES might be possible 

Anyway thank you for the plugin as it is now I used it a lot. 
Here is the broadcast link

https://api.tornelo.com/divisions/24090802-f209-4106-ae35-a92f7f2824e4/broadcast-pgn?convertMatchNumbers=false&token=4590c1a238a3cc40cd66339d0192c3c7&withClocks=true

Regards,  
Helmer

**Comment by yo35** [comment]
Hi,
Live broadcast is not the primary target of RPB Chessboard. But I'm not sure it is the primary target of WordPress either, which is meant to be a blog platform in the first place.

Still, with `[pgn url=<url-to-the-broadcast-link> game=<index-of-the-game>]`, it should be possible to display the (live) games in a page or post. I didn't try, but it should work out of the box. If you add an auto-refresh mechanism on top of that (look for "wordpress auto refresh page" in google), the page will receive automatic updates during the game.

Of course, this is not perfect, it would be better to receive an active update notification from the broadcast stream instead of polling it with a predefined time-step (especially if you want to broadcast fast games such as blitz/bullet), but implementing such active notification mechanism would require some specific work.

Regarding the multi-board feature: it's possible to have several RPB Chessboard games within the same page. It's up to you to layout the page in way you intent to.

**Comment by schaakverenigingmaastricht** [comment]
Thanks for the prompt response. 

1. When I download the pgn and add the text manually in a post/page it displays  corectly only **one**  game while there are more games in the file  (there is a syntax error in the file  because using zero 's for castiling but this will be fixed by us) 

I  need to add a [/pgn]  to show multiple games could your plugin allow multiple games and for instance assume [Event] as a divider? 

2. I tried this [pgn url = "https://api.tornelo.com/divisions/24090802-f209-4106-ae35-a92f7f2824e4/broadcast-pgn?convertMatchNumbers=false&token=4590c1a238a3cc40cd66339d0192c3c7&withClocks=true" ]
Two issues: 
a. An error is returned
"Cannot download the PGN file.
https://api.tornelo.com/divisions/24090802-f209-4106-ae35-a92f7f2824e4/broadcast-pgn?convertMatchNumbers=false&amp;token=4590c1a238a3cc40cd66339d0192c3c7&amp;withClocks=true"

b. I don't know to deal with game=index of game 

I think that if it would work it would still be a nice option.  
Can you give some hints how to get it working?

**Comment by schaakverenigingmaastricht** [comment]
**Side topic:** 
 I read this on wikipedia: 
" SAN kingside castling is indicated by the sequence O-O; queenside castling is indicated by the sequence O-O-O (note that these are capital Os, not zeroes, contrary to the FIDE standard for notation)."

I did not know this... It is annoying that FIDE for a reason, unknown to me, deferred from the orginal standard. 
My question would you be willling to support the zero as well in your plugin?

**Comment by yo35** [comment]
Hi,
I've tried to retrieve a PGN from the URL you mentioned above, and it appears to work on my test website. In the admin, I've created a post as follow:

![tornello1](https://user-images.githubusercontent.com/5740068/108911496-9a550f80-7627-11eb-880e-939c02a714db.png)

... which has generated the following post:

![tornello2](https://user-images.githubusercontent.com/5740068/108911586-b789de00-7627-11eb-8ea7-c2646ea1ff22.png)

The first game reports an error, but as you've noticed, this is due to the fact that the retrieved PGN file does not obey to the PGN format specification with respect to castle encoding. The second game is fine. No data access issue.

`game=index` is meant to select which game is supposed to be displayed among the games found in the PGN file. `game=0` displays the first game, `game=1` the second one, etc... There is a feature request to make RPB chessboard capable of displaying multiple games - with selection let to the user through dropdown menu - but this is not available yet. See #168 and #93 for more details.

The problem ill-formed PGN files, that encode castling moves with zero instead of capital O has been discussed in #142.
